src/print/mod.rs

```diff
@@ -33,6 +34,45 @@ use std::path::PathBuf;
 
 use cargo_metadata::Package;
 
+/// The result of rendering a template (main.wxs, License.rtf, ...)
+pub struct RenderOutput {

```

Comment:
Should this eventually implement the `std::io::Write` trait?

We want to write either to a String, STDOUT, or a File for most output. The `std::io::Cursor`could be used for writing to a String, the `std::io` "stuff" could be used for STDOUT or a File. Again, coming in cold and trying to wrap my brain around this.

Reply:
Hmm not quite sure what you're trying to describe -- the RenderOutput type is really intended to just be a String, but I wanted to get rid of a place where two parts of the code have to assume the location they're talking about.

Basically when rendering out a template we really need to know where on disk that file will be written, because templates contain relative paths. So since the template renderer already has to know the final path, I decided it was best for it to report back that path so that the caller didn't have to assume/know anything, and could just faithfully use the reported path.

And then once it already stored the path I was like "well ok I can just have a method to write it to disk". There is however an outstanding issue -- as designed, the "print to stdout" mode definitionally has no know target path, and can only output one file (when we may need to output up to 3 (well really only 2))... not sure how to make that work.

Reply:
I was more or less just "thinking out loud" while I was reviewing the code. I was wondering if the `std::io` already had some functionality that could be reused but I was misremembering from when I was trying to have output be either STDOUT or a file and there were some traits I had to play around with to make it all work.